### PR TITLE
ServicesWatcher 1.0 recipe

### PR DESCRIPTION
**Pull request by Perelandra0x309**
Request to add a recipe for ServicesWatcher 1.0 which I have added to GitHub,

### CODE REVIEW

File: haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+these services will automatically update as any of them quit or start.  You \
+may manually start, stop and restart these services on demand."
+HOMEPAGE="https://github.com/Perelandra0x309/services-watcher"
+COPYRIGHT="2010-2016 Brian Hill"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/Perelandra0x309/services-watcher/archive/${portVersion}_source.tar.gz"
+CHECKSUM_SHA256="d624c65ae4c55447a4778366e6a9267fe2c44eceec2834081e28104be0827a0b"
+SOURCE_DIR="services-watcher-${portVersion}_source"
+ARCHITECTURES="x86_gcc2 !x86"
```
Comment (fbrosson) [inline haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe]: ARCHITECTURES likes to have an empty line above it ;-)

File: haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+PROVIDES="
+	serviceswatcher = $portVersion
+	app:ServicesWatcher = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+BUILD_REQUIRES="
```
Comment (fbrosson) [inline haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe]: BUILD_REQUIRES likes to have an empty line above it ;-)

**Comment by humdingerb** [comment]
This PR includes old commits that were already merged for Einsteinium. Does anyone know if github's "Squash & merge" takes care of that correctly?

@Perelandra0x309 : I think you'll need to update your checked out fork, see [How do I sync my master branch with original repository?](https://github.com/haikuports/haikuports/wiki/DevelopmentModel) in the wiki.

If you don't already, also using feature branches is a good idea, see that same wiki page.

**Comment by Perelandra0x309** [comment]
Humdinger, yes I tried to follow the wiki and this is what I ran:
git status
git branch
git remote add upstream https://github.com/haikuports/haikuports.git
git pull upstream master
git status

Maybe I should delete my fork and start a fresh one?

**Comment by humdingerb** [comment]
I'm very bad at rescuing a botched repos, so I'd also start with a clean slate... :)
Then proceed as the DevelopmentModel wiki suggests.

Who dares to "squash & merge" this PR? :)

**Comment by fbrosson** [comment]
> This PR includes old commits that were already merged for Einsteinium. Does anyone know if github's "Squash & merge" takes care of that correctly?

@humdingerb: I _think_ it does, but I'm not sure :-(

@Perelandra0x309: A few months ago GitHub has added a feature that lets you change the head branch associated to a PR. I think it requires you to delete the branch and then to click on the PR and select the branch you want to attach to it. But here, since the head branch is master, you might not be able to delete it, so the web-based method will probably not work for this PR.

I see 2 methods that you might follow and that do not require to close this PR (and open another one.)

First method: check the "Allow edits from maintainers" (on the bottom-right corner of the page showing your PR) and let one of us put your recipe in a commit. If you wish I can even keep you as the author of the commit ;-)

Second method:
1) backup your recipe to a safe place.
2) drop your local clone.
3) Run these commands:
```
git clone https://github.com/haikuports/haikuporter.git
cd haikuports
git remote set-url origin https://github.com/Perelandra0x309/haikuports.git
git remote add upstream https://github.com/haikuports/haikuports.git
git checkout -b ServicesWatcher master

mkdir -p haiku-apps/serviceswatcher
cp  /path/to/the/backup/dir/serviceswatcher-1.0.recipe haiku-apps/serviceswatcher
git commit haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe
git push --set-upstream origin ServicesWatcher

curl --user Perelandra0x309 --request PATCH \
     --data '{"issue":"807", "head":"Perelandra0x309:ServicesWatcher", "base":"master"}' \
     https://api.github.com/repos/haikuports/haikuports/pulls
```
(then type the password of you GitHub account when prompted by curl)

Important: I never tried this method, so I don't know if it really works...

**Comment by Perelandra0x309** [comment]
@fbrosson OK I tried your git kung-fu :-) But with 2 changes, first line cloned the haikuports.git, not the haikuporter.git, and did a 'git add haiku-apps/serviceswatcher/serviceswatcher-1.0.recipe' before the commit.
I got through it all but with this error on the curl command:

~/myforks/haikuports> curl --user Perelandra0x309 --request PATCH --data '{"issue":"807", "head":"Perelandra0x309:ServicesWatcher", "base":"master"}' https://api.github.com/repos/haikuports/haikuports/pulls
Enter host password for user 'Perelandra0x309':
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}

I also tried to edit the pull request on the website to change the brach it pulls from, but it doesn't allow me to select another.

**Comment by fbrosson** [comment]
Oops, it looks like it's harder than I though...

Oh, I've found the pages I had seen a few weeks ago:
* https://stackoverflow.com/questions/10081053/how-to-change-the-base-branch-of-a-pull-request
* https://github.com/blog/2224-change-the-base-branch-of-a-pull-request
* https://help.github.com/articles/changing-the-base-branch-of-a-pull-request/

Have you tried the Edit button the way they suggest?

**Comment by fbrosson** [comment]
Oops, my mistake. it only allows to change the **_base_** branch, not the **_head_** branch.

**Comment by fbrosson** [comment]
If your newly created ServicesWatcher branch is OK (against upstream/master) then this might work:
```
git checkout ServicesWatcher
git push -f --set-upstream origin master
```

**Comment by Perelandra0x309** [comment]
@fbrosson Haha, OK those two commands closed it!  I think time to start fresh...

**Comment by fbrosson** [comment]
Oops, sorry. I guess the only thing to do now is open a new PR.

**Comment by fbrosson** [comment]
Wait! the "Reopen and comment" might become clickable.

**Comment by fbrosson** [comment]
I wonder what would happen if you add your commit to your local master and then push it to your remote master.
The "Reopen and comment" button below is disabled, but if I put the mouse over it then a message appears: "There are no new commits on the Perelandra0x309:master branch."
So, if you would like to try, for this PR, to push a commit to your remote master, we might be able to reopen this PR and then merge it.

**Pull request closed without merging**